crafting-dead-core/src/main/java/com/craftingdead/core/world/action/item/ItemAction.java

```diff
@@ -101,8 +105,8 @@ protected Optional<Item> getResultItem(LivingExtension<?, ?> performer) {
 
   @Override
   public boolean tick() {
-    if (!this.getPerformer().getLevel().isClientSide()
-        && !this.getPerformer().getEntity().isUsingItem()) {
+    if (!this.getPerformer().getEntity().isUsingItem() || !compareStacks(this.originalStack,

```

Comment:
Can't you just compare the item instance? It should remain the same throughout usage (Minecraft does that in `LivingEntity::updatingUsingItem`).

Reply:
I was unsure if NBT check would be needed. Switched to comparing the item directly

Reply:
I meant compare the `ItemStack` instance so just `oldStack == newStack`